### ISSUE TITLE
Any chance you'll be putting the Website with the APK back up?

### ISSUE THREAD

**Issue by Rajvir8** [MathewSachin/Fate-Grand-Automata#533]
I can't update from Google Play Store due to installing from an APK sadly, and I've noticed you aren't exactly trying to keep the GP one up to date compared to Github as well.

Sadly the option to export Scripts is broken for me at the moment.

I guess I should also look into how to just download the code and turn it into an APK as well to be honest.  Just not sure where to look or what I'd have to install to do so.

**Comment by reconman** [comment]
You can still download apks from builds here https://github.com/MathewSachin/Fate-Grand-Automata/actions?query=branch%3Amaster+event%3Apush

**Comment by MathewSachin** [comment]
I posted on Gamepress about canary builds.
https://github.com/MathewSachin/Fate-Grand-Automata/releases

**Comment by MathewSachin** [comment]
> Sadly the option to export Scripts is broken for me at the moment

Broken, as in?

You won't be able to update to the PlayStore without uninstalling old version first since APK signature changed.
And, the Canary build will install alongside your already installed version.

**Comment by Rajvir8** [comment]
This is perfect thanks, sorry about the delay ended up busy and didn't check my email for two days.

Went to releases and Grabbed 2.0

Installing it now and it updated perfectly.

Thanks again.

**Comment by Rajvir8** [comment]
Also installed the Canary Build but it sadly doesn't have my scripts.  I'll just copy and paste I guess as well.  the folder and hope it gives me everything after double checking settings

**Comment by Rajvir8** [comment]
Also just noticed the Gift Box Present collection counting, and god that's amazing.

It'll actually make stacking so much more feasible and less annoying.

There isn't a filter for showing Silver EXP only in NA though only JP?  Would we still have to manually collect any Silver EXP that's in two stacks or higher?

**Comment by Rajvir8** [comment]
I'll probably stick to the normal one for now but I am extremely interested in Skill Spam.

Is it possible that it will work similarly to NP Spam when mixed with skills you choose?

As in it will only use the skills you want it to use until it hits the end of your list at which point it will automatically skill spam every time it sees a skill is available.

Out of curiosity can it detect your MC Skills?

Also much more importantly how does skill spam deal with targetable skills? Does it just auto use it on Servant 1 or something?

**Comment by Rajvir8** [comment]
The one thing I dislike about the Canary so far compared to the normal one would probably be writing the skill menu, so far I can't figure out how to be allowed to just type in my own code again, other than exporting it and then importing it.

Instead it wants to force me into the Menu where it writes the code for me and I just choose what Servant skills I want to use.

I can understand why that'd be good for a lot of people, but I prefer writing my code on a notepad thanks to being able to easily look at my Servant Skills and then copy paste it over.

**Comment by MathewSachin** [comment]
> Had to import one at a time but glad I managed it.

In the default Android file picker, there is a 'Show Internal Storage' option from the dropdown menu. It's weird but that's how Android is.
Also, you can long-press to pick multiple files in a single Import.

> Would we still have to manually collect any Silver EXP that's in two stacks or higher

Silver EXP is always picked irrespective of stack size.

For how Skill Spam works, see description of #511

> Out of curiosity can it detect your MC Skills?

No, it can't right now. I didn't want it to open Master Skills menu every turn to check.

> The one thing I dislike about the Canary so far compared to the normal one would probably be writing the skill menu, so far I can't figure out how to be allowed to just type in my own code again, other than exporting it and then importing it.

It's not canary, the PlayStore builds are the same. You were on a much older version I guess.
There's a `Skill command text box` option in `More options`.

**Comment by Rajvir8** [comment]
> In the default Android file picker, there is a 'Show Internal Storage' option from the dropdown menu. It's weird but that's how Android is.
> Also, you can long-press to pick multiple files in a single Import.

Sorry not used to default android, and man I wish I had figured that out.  I did try long press as I did use that for deleting multiple of them afterwards.  Must have messed that one up to somehow.  But hey it worked out.

> Silver EXP is always picked irrespective of stack size.
That's perfect, looking forward to using in the next lottery.

> For how Skill Spam works, see description of #511
Huh that's honestly rather interesting, I'll certainly turn it on then.  A bit of a shame it won't use the skills of a new Servant coming in as that'd probably be a bigger difference than doing it when they come off cooldown.

Overall though, the only thing I'd lose at worst would be a chance for some skills to be pressed, as due to targeting it'll more often have at least some skills being wrong which can potentially change turn count and more.  

This will be especially a massive quality of life improvement for my Caster Stall Teams for those pesky all Assassin set ups.

And overall allows me to use a lot more skills early on like in normal gameplay.  I often save long skills with good cooldowns for last anyway as I really want Gil's charisma for example during final wave as well.  This allows me to cast them far more freely, same with Waver's cooldowns.  Be that much more likely to have there NP ready when I want it.

> No, it can't right now. I didn't want it to open Master Skills menu every turn to check.
Completely understandable, I was just curious how you'd program that if it did work.  

> It's not canary, the PlayStore builds are the same. You were on a much older version I guess.
I must have been, I was on Version 2.0 the most up to date one on the website I think before it got taken down.

> There's a `Skill command text box` option in `More options`.
Found it and enabled it, thanks again for all the help and explanations.  It really helped.

While your here  I noticed it's ditched the different modes options like Lottery/Friendship rolling

Does it just automatically detect that now when it's in the correct Menu?

**Comment by Rajvir8** [comment]
Also is there any way to set NP Spam and Skill Spam to be automatically on?  I could see myself manually turning it off in a few scenario's but most times I'd want it to be on.

If not I guess I'll just have to remember when making new ones like always clicking use Friend.

I certainly have to go to all my old ones and add Spam NP and Spam Skills

**Comment by Rajvir8** [comment]
Following skill spam is there any chance NP Spam could remember the order you put it in?  In which case if it's 645 it will always click on third servant's NP Location first?

**Comment by MathewSachin** [comment]
> Does it just automatically detect that now when it's in the correct Menu?

yeah, you'll notice a new script when clicking on PLAY button which will do that.

> Also is there any way to set NP Spam and Skill Spam to be automatically on? I could see myself manually turning it off in a few scenario's but most times I'd want it to be on.

No, not everyone wants NP/Skill spam since most people 3T, so you'll have to set in individual scripts

> A bit of a shame it won't use the skills of a new Servant coming in

Not every servant has skills you want to use either. Like the ones with demerits like TamaLancer. So, it only uses the ones you told it to.

> Following skill spam is there any chance NP Spam could remember the order you put it in? In which case if it's 645 it will always click on third servant's NP Location first?

I'll have to look into it, but you should open a new Feature request for that.

**Comment by Rajvir8** [comment]
> Not every servant has skills you want to use either. Like the ones with demerits like TamaLancer. So, it only uses the ones you told it to.

True, the vast majority of cases I feel them just dropping there skills would be worth it, and if it was a toggle I'd 100% turn it on as the vast majority of Servant's aren't like that.  Even for TamaLancer I wouldn't mind if Stun wasn't particularly bad in auto if you had an NP.  As for my old version that would straight up cause it to forever pause there.

Still considering how difficult it'd probably be to program I can't exactly disagree and skill spam seems better as it currently is compared to the old system anyways so. (Shrugs)



> yeah, you'll notice a new script when clicking on PLAY button which will do that.

Perfect thanks, I thought I saw one when I opened it in FGO, but when I opened it by itself it just brought up all my battle commands.


> No, not everyone wants NP/Skill spam since most people 3T, so you'll have to set in individual scripts

That honestly always surprises me, I've set up plenty of 3T before and I'm fine with the spam's.

They only kick in AFTER you've used all of your commands after all, so both Spams should only kick in after Auto has completed your 3T.  Thus to me it's always good to have just in case something goes wrong, such as the NP damage leaning towards the low end or something else and potentially failing.

Still I guess I'll just have to remember, thanks for the explanation as always.




> I'll have to look into it, but you should open a new Feature request for that.

Will do thanks.
